napari/layers/shapes/_shapes_mouse_bindings.py

```diff
@@ -224,20 +236,17 @@ def vertex_insert(layer, event):
     else:
         new_type = None
     closed = shape_type != Path
-    vertices = layer._data_view.displayed_vertices[
-        layer._data_view.displayed_index == index
-    ]
+    vertices = layer._data_view.shapes[index].data
     if not closed:
         if int(ind) == 1 and loc < 0:
             ind = 0
         elif int(ind) == len(vertices) - 1 and loc > 1:
             ind = ind + 1
 
     # Insert new vertex at appropriate place in vertices of target shape
-    vertices = np.insert(vertices, ind, [layer.displayed_coordinates], axis=0)
+    vertices = np.insert(vertices, ind, [layer.coordinates], axis=0)
```

Comment:
I don't understand why this change works at all?

Reply:
It's only possible because we also changed the line above `vertices = layer._data_view.shapes[index].data`. Basically we never drop down to the 2D representation, so we can always stay in the nD representation